packages/query/src/block-processor.ts

```diff
@@ -388,4 +409,46 @@ export class BlockProcessor implements BlockProcessorInterface {
       await this.indexedDb.upsertValidatorInfo(validatorInfoResponse.validatorInfo);
     }
   }
+
+  private async updatePrices(swapOutputs: BatchSwapOutputData[], height: bigint) {
+    for (const swapOutput of swapOutputs) {
+      const swapAsset1 = getSwapAsset1(swapOutput);
+      const swapAsset2 = getSwapAsset2(swapOutput);
+
+      let numerairePerUnit = 0;
+      let pricedAsset: AssetId | undefined = undefined;
+
+      // case for trading pair <pricedAsset,numéraire>
```

Comment:
We cannot know in advance the order of assets in a trading pair, so we must handle both cases